VpNet/ManagedApi/Scene/VpObject.cs

```diff
@@ -1,20 +1,188 @@
+        }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != this.GetType()) return false;
+            return Equals((VpObject) obj);
+        }
+
+        /// <inheritdoc />
+        public override int GetHashCode() => Id;
+
+        /// <summary>
+        ///     Modifies the object.
+        /// </summary>
+        /// <param name="builder">The builder of the object to modify.</param>
+        public async Task ModifyAsync(Action<VpObjectBuilder> builder)
+        {
+            await Task.Run(() =>
+            {
+                var model = new VpObjectBuilder();
+                builder(model);
+                lock (_client)
+                {
+                    IntPtr handle = _client.NativeInstanceHandle;
+                    if (!(model.Action is null))
+                    {
+                        vp_string_set(handle, StringAttribute.ObjectAction, model.Action);
+                    }
+
+                    if (!(model.Description is null))
+                    {
+                        vp_string_set(handle, StringAttribute.ObjectDescription, model.Description);
+                    }
+
+                    if (!(model.Model is null))
+                    {
+                        vp_string_set(handle, StringAttribute.ObjectModel, model.Model);
+                    }
+
+                    if (!(model.Type is null))
+                    {
+                        vp_int_set(handle, IntegerAttribute.ObjectType, model.Type.Value);
+                    }
+
+                    if (!(model.Angle is null))
+                    {
+                        vp_double_set(handle, FloatAttribute.ObjectRotationAngle, model.Angle.Value);
+                    }
+
+                    if (!(model.Position is null))
+                    {
+                        Vector3 position = model.Position.Value;
+                        vp_double_set(handle, FloatAttribute.ObjectX, position.X);
+                        vp_double_set(handle, FloatAttribute.ObjectY, position.Y);
+                        vp_double_set(handle, FloatAttribute.ObjectZ, position.Z);
+                    }
+
+                    if (!(model.Rotation is null))
+                    {
+                        Vector3 rotation = model.Rotation.Value;
+                        vp_double_set(handle, FloatAttribute.ObjectRotationX, rotation.X);
+                        vp_double_set(handle, FloatAttribute.ObjectRotationY, rotation.Y);
+                        vp_double_set(handle, FloatAttribute.ObjectRotationZ, rotation.Z);
+                    }
+
+                    if (!(model.Data is null))
+                    {
+                        SetData(handle, DataAttribute.ObjectData, model.Data);
+                    }
+
+                    int reason = vp_object_change(handle);
```

Comment:
Calling vp_object_change while only having set some of the object properties does not make it only update those fields. Instead, it will use the values that were still in the fields that were not set (which could be from a query, or a previous object add/load/change).

To be safe you will have to fill all the object fields. It's probably a better idea to use `VirtualParadiseClient.ChangeObjectAsync` so you also get the async handling of object change.